leakcanary-android-proguard/src/main/java/com/squareup/leakcanary/proguard/CopyProguardMappingFileTask.kt

```diff
@@ -0,0 +1,48 @@
+package com.squareup.leakcanary.proguard
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+@CacheableTask
+open class CopyProguardMappingFileTask : DefaultTask() {
+
+  @Input
+  var variantDirName: String? = null
+
+  @Input
+  @Optional
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mappingFile: File? = null
+
+  @Input
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mergeAssetsDirectory: File? = null
+
+  @get:OutputFile
+  @get:PathSensitive(PathSensitivity.RELATIVE)
+  val leakCanaryAssetsOutputFile: File
+    get() = File(mergeAssetsDirectory, "leakCanaryProguardMapping.txt")
+
+  init {
+    description = "Puts proguard mapping file in assets directory."
+  }
+
+  @TaskAction
+  fun copyProguardMappingFile() {
+    mappingFile?.let { mappingFile ->
+      if (mappingFile.exists()) {
+        if (leakCanaryAssetsOutputFile.exists()) {
+          leakCanaryAssetsOutputFile.delete()
+        }
+        mappingFile.copyTo(leakCanaryAssetsOutputFile)
```

Comment:
Are you sure the parent folder always exists ? 
If not, you might need to create the parent dir tree of the destination.

Reply:
Good point!